airflow/models/dag.py

```diff
@@ -473,6 +475,8 @@ def __init__(
         validate_key(dag_id)
 
         self._dag_id = dag_id
+        self._display_name = display_name or dag_id
```

Comment:
Since we already have fallback logic in the property, the `or dag_id` part is not needed. It’s better to be able to store the arguments losslessly.